lib/spack/spack/ci.py

```diff
@@ -613,7 +613,7 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
                     debug_flag = '-d '
 
                 job_scripts = [
-                    'spack env activate .',
+                    'spack env activate --without-view .',
```

Comment:
@becker33 It seems like this was encountered by @haampie in attempts to run compiler bootstrapping pipelines, though I have not encountered any need for it in normal pipelines.  Can you shed any light on why it might be necessary?

Reply:
If one of the dependencies of a package is `python`, spack seems to be confusing the system python with the python version in the view.

I encountered it when trying to build `hpx@1.4.1` in one of the lasts steps (well, that's where python gets installed...). The main issue is that it cannot find botocore anymore, probably the search paths are simplify confused by loading the other python into a view.

Reply:
So you needed this independent of compiler bootstrapping?  I'm just surprised because I have run lots of pipelines with packages that have a dependency on `python`.

Reply:
Hm, I'm running it in a non-standard container runtime, so it could be that that somehow influences the environment a bit. Let me try and see tomorrow if it's reproducible when dropping this commit ^ but keeping all the other current patches that seem necessary.

Reply:
@haampie are you running this on Spack v0.14.2? We had a bug up until v0.15.0 that caused Spack to fail if an environment including python was partially installed and the view was activated. If you're seeing this behavior in v0.15.0 that is a much bigger concern to me.

On the other hand, there's not really any reason to add the view to the user environment as part of the CI job, so eliminating it should have a (probably negligible) performance benefit.